### PR TITLE
Add rpm support for integTest framework

### LINKED ISSUE #1951: Automate integ test using RPM bundle.
Currently our integ tests are working only for using tarball distribution. We now have workflow to generate RPM bundle. We need incorporate the automated integ tests for RPM bundle.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add rpm support for integTest framework
 
### Issues Resolved
#1951
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/system/process.py [hunk 1]
```diff
@@ -12,12 +12,14 @@
 
 
 class Process:
-    def __init__(self) -> None:
+    def __init__(self, filename: str, distribution: str) -> None:
```
Comment (dblock) [inline src/system/process.py:15]: Process is a generic class in `system`, it should have nothing to do with distribution, this doesn't belong here.

File: src/system/process.py [hunk 2]
```diff
@@ -65,6 +67,10 @@ def terminate(self) -> int:
         self.return_code = self.process.returncode
         self.process = None
 
+        if self.distribution == "rpm":
```
Comment (dblock) [inline src/system/process.py:70]: This should not live in `Process`.

File: src/test_workflow/integ_test/service.py [hunk 3]
```diff
@@ -19,14 +20,30 @@ class Service(abc.ABC):
...
+        self.filename = filename
         self.work_dir = work_dir
         self.version = version
+        self.distribution = distribution
         self.security_enabled = security_enabled
         self.additional_config = additional_config
         self.dependency_installer = dependency_installer
 
-        self.process_handler = Process()
+        self.process_handler = Process(self.filename, self.distribution)
```
Comment (dblock) [inline src/test_workflow/integ_test/service.py:32]: This class is a generic `Service`. Subclass it into `ServiceOpenSearchTar` and `ServiceOpenSearchRpm`, `ServiceOpenSearchDashboardsTar` etc.,  and implement the differences in the child classes.

File: src/test_workflow/integ_test/service_opensearch.py [hunk 4]
```diff
@@ -50,11 +56,31 @@ def __download(self):
...
         logging.info(f"Downloaded bundle to {os.path.realpath(bundle_name)}")
 
-        logging.info(f"Unpacking {bundle_name}")
-        with tarfile.open(bundle_name, 'r') as bundle_tar:
-            bundle_tar.extractall(self.work_dir)
-
-        logging.info(f"Unpacked {bundle_name}")
+        logging.info(f"Installing {bundle_name} in {self.install_dir}")
+
+        if self.distribution == "tar":
```
Comment (dblock) [inline src/test_workflow/integ_test/service_opensearch.py:61]: Subclass `ServiceOpenSearch` into `ServiceOpenSearchTar` and ``ServiceOpenSearchRpm` and implement the differences there.

**Review by dblock — CHANGES_REQUESTED** [review]
This PR bolts the changes about specifics on OpenSearch distributions all the way down in base classes that are reusable and generic. Refactor this out into child classes.

**Comment by peterzhuamazon** [comment]
Replaced by #2000.

After talking to @dblock I will re-write this PR so proper functions will be sourced from corresponding classes.

Thanks.

**Pull request closed without merging**